src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1943,6 +1943,110 @@ internal void IncrementPasteOffset()
             this.currentPasteOffset = (this.currentPasteOffset + PasteOffsetStep) % PasteOffsetMax;
         }
 
+        #region [Nodes Info]
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings
+        /// </summary>
+        public bool HasWarnings
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Warning || n.State == ElementState.PersistentWarning); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings with no Geometry
+        /// </summary>
+        public bool HasWarningsWithNoGeometry
+        {
+            get { return Nodes.Any(n => (n.State == ElementState.Warning || n.State == ElementState.PersistentWarning) && !n.Category.StartsWith("Geometry.")); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if workspace run with errors
+        /// </summary>
+        public bool HasErrors
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Error); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if home workspace is displayed with infos
+        /// </summary>
+        public bool HasInfos
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Info); }
+        }
+
+        /// <summary>
+        /// All the Nodes identified as deprecated
+        /// </summary>
+        public IEnumerable<NodeModel> DeprecatedNodes
+        {
+            get
+            {
+                return Nodes.Where(node => node.NodeInfos.Any(nodeInfo => nodeInfo.Message.Contains("This method is deprecated and will be removed")));
```

Comment:
Can we not check if these nodes have a warning or info type or ID that is "deprecated" or some such type? If not, we should use the resource string instead of the hardcoded string here to check for the message.

Reply:
> Can we not check if these nodes have a warning or info type or ID that is "deprecated" or some such type? If not, we should use the resource string instead of the hardcoded string here to check for the message.

As far as I know this is the only way to detect it in this context, otherwise please let me know.